src/fut/stream_finish.rs

```diff
@@ -1,21 +1,24 @@
 use std::pin::Pin;
-
-use futures_util::task::{Context, Poll};
+use std::task::{Context, Poll};
 
 use crate::actor::Actor;
 use crate::fut::{ActorFuture, ActorStream};
 
-/// A combinator used to convert stream into a future, future resolves
-/// when stream completes.
-///
-/// This structure is produced by the `ActorStream::finish` method.
-#[must_use = "streams do nothing unless polled"]
-#[pin_project::pin_project]
-#[derive(Debug)]
-pub struct StreamFinish<S: ActorStream>(#[pin] S);
+pin_project_lite::pin_project! {
+    /// A combinator used to convert stream into a future, future resolves
+    /// when stream completes.
+    ///
+    /// This structure is produced by the `ActorStream::finish` method.
+    #[must_use = "streams do nothing unless polled"]
+    #[derive(Debug)]
+    pub struct StreamFinish<S: ActorStream> {
+        #[pin]
+        stream: S
+    }
+}
 
-pub fn new<S: ActorStream>(s: S) -> StreamFinish<S> {
-    StreamFinish(s)
+pub fn new<S: ActorStream>(stream: S) -> StreamFinish<S> {
```

Comment:
I personally feel this style is clearer in docs for simple bounds.

```suggestion
pub fn new(stream: impl ActorStream) -> StreamFinish<S> {
```

Reply:
Oh derp... sorry, S is used actually.